Overwrite sigma of prior distributions.

toshihikoyanase opened the pull request:
This PR changes the behavior of TPESampler.
Currently, the sigma values of prior distributions are changed depending on the observations.
In this change, the sigma values are overwritten by a predetermined value, similarly to the TPE implementation of hyperopt.

fofof200 commented:
This patch looks not working, if consider_priror is False. And it is capable to set consider_prior as False.

toshihikoyanase commented:
> This patch looks not working, if consider_priror is False. And it is capable to set consider_prior as False.

Thank you for your comment. I fixed the problem and added unit tests to confirm `ParzenEstimator._calculate` works with both `consider_prior=True` and `consider_prior=False`. The test coverage is not enough, but we'll improve the tests in future PR.

g-votte commented on tests/samplers_tests/tpe_tests/test_parzen_estimator.py:
```diff
@@ -0,0 +1,85 @@
+from optuna.samplers.tpe.sampler import default_weights
+
+
+class TestParzenEstimator(object):
```
As there're many duplicated logics b/w the two test cases, could you extract them to a common one? (E.g., with `pytest.mark.parametrize`.)

toshihikoyanase replied:
Thank you for your suggestion. As you mentioned, I used `pytest.mark.parametrize` to remove the duplication. Any feedback is welcome.
